pallet/src/tests.rs

```diff
@@ -1,7 +1,57 @@
 #![cfg(test)]
 
-use crate::mock::*;
+use crate::{migrations, mock::*, pallet};
 use frame_support::assert_ok;
+use frame_support::traits::{GetStorageVersion, StorageVersion};
+
+#[test]
+fn migration_from_v0_to_v1_works() {
+    new_test_ext(&[(1u64, 1u64), (2u64, 2u64)]).execute_with(|| {
+        frame_support::migration::put_storage_value(
+            b"Aleph",
+            b"Validators",
+            &[],
+            vec![AccountId::default()],
+        );
+
+        frame_support::migration::put_storage_value(
+            b"Aleph",
+            b"SessionForValidatorsChange",
+            &[],
+            1u32,
+        );
+
+        let v0 = <pallet::Pallet<Test> as GetStorageVersion>::on_chain_storage_version();
+
+        assert_eq!(
+            v0,
+            StorageVersion::default(),
+            "Storage version before applying migration should be default"
+        );
+
+        let _weight = migrations::v0_to_v1::migrate::<Test, Aleph>();
+
+        let v1 = <pallet::Pallet<Test> as GetStorageVersion>::on_chain_storage_version();
+
+        assert_eq!(
+            v1,
+            StorageVersion::new(1),
```

Comment:
surely it is not the case, but I would be more explicit with ensuring that `v0 != v1`, either by a direct comparison or by using `StorageVersion::new(0)` instead of `StorageVersion::new(1)`